test/test_distributions.py

```diff
@@ -4092,5 +4092,121 @@ def tearDown(self):
         super(TestCase, self).tearDown()
         Distribution.set_default_validate_args(False)
 
+
+class TestJit(TestCase):
+    def _examples(self):
+        for Dist, params in EXAMPLES:
+            for param in params:
+                print('testing {}'.format(Dist.__name__))
+                keys = param.keys()
+                values = tuple(param[key] for key in keys)
+                if not all(isinstance(x, torch.Tensor) for x in values):
+                    continue
+                sample = Dist(**param).sample()
+                yield Dist, keys, values, sample
+
+    def test_sample(self):
+        for Dist, keys, values, sample in self._examples():
+            if Dist in [Geometric, Cauchy, StudentT]:
+                continue  # FIXME missing support for aten::uniform, aten::cauchy
+
+            def f(*values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.sample()
+
+            torch.jit.trace(f, values, check_trace=False)
```

Comment:
While trace might succeed in most cases, it doesn't mean that the trace is correct. In particular the outputs could be embedded as constants. It would be better to use `torch.random.fork_rng` to ensure that the traced function returns the same outputs as `f`, and that they are different every time you run it. Additionally, you might want to do something like:
```python
self.assertTrue(any(n.isNondeterministic() for n in traced_f.graph.nodes()))
```